### PR TITLE
[sysdig-deploy][node-analyzer][kspm-collector][rapid-response] Adding deployment tests and serviceaccount for rapid-response.

### PR DESCRIPTION
**Pull request by yashgiri**
## What this PR does / why we need it:

This PR adds deployment tests that test the rollout status of all the charts after they are installed.
This PR included adding service account for rapid-response chart.

## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Title of the PR starts with chart name (e.g. [mychartname])
- [x] Chart Version bumped for the respective charts
- [x] Changelog updated for the charts with version updates.
- [x] Variables are documented in the README.md (or README.tpl in some charts)
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [ ] All test files are added in the tests folder of their respective chart and have a "_test" suffix

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

File: charts/rapid-response/templates/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - ''
+      - 'rbac.authorization.k8s.io'
+      - 'extensions'
+      - 'apps'
+      - 'batch'
+      - 'networking.k8s.io'
+      - 'autoscaling'
+      - 'policy'
+      - 'storage.k8s.io'
+    resources:
```
Comment (ironashram) [inline charts/rapid-response/templates/clusterrole.yaml]: does it need all these resources?

Comment (yashgiri) [reply charts/rapid-response/templates/clusterrole.yaml]: I am not sure at the moment but keeping it consistent with the other charts. If needed an evaluation can be done in some other ticket on what all would be required.

**Review by ironashram — APPROVED** [review]

**Comment by yashgiri** [comment]
Hi @dark-vex can I get your review on this?

File: charts/rapid-response/templates/clusterrole.yaml (hunk 1 above)
Comment (dark-vex) [reply charts/rapid-response/templates/clusterrole.yaml]: @yashgiri @ironashram Actually the rapid response doesn't require those resources, it just need his daemonset+configmaps/secrets
https://docs.sysdig.com/en/docs/installation/rapid-response-installation/#as-kubernetes-daemonset

File: charts/sysdig-deploy/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,16 @@
...
 
+## 1.5.9
+### Minor changes:
+* kspm-collector
+    * Add deployment test to kspm collector chart
+* node-analyzer
+    * Add deployment test to node analyzer chart
+* rapid-response
+    * Add deployment test to rapid response chart
+    * Adding serviceaccount template
```
Comment (lilx1ao) [inline charts/sysdig-deploy/CHANGELOG.md]: I feel this should be on a separate pr (next time). maybe add this into the pr title?

File: charts/rapid-response/templates/clusterrole.yaml (hunk 1 above)
Comment (yashgiri) [reply charts/rapid-response/templates/clusterrole.yaml]: cool i will reduce it down to those 2 and if I think anything is failing I will add anything required.

File: charts/sysdig-deploy/CHANGELOG.md (hunk 2 above)
Comment (yashgiri) [reply charts/sysdig-deploy/CHANGELOG.md]: Hi, @lilx1ao Since my changes to check rollout status gets permission issues without a service account both the changes are needed in the same PR or it will get blocked during the CI stage.
Edit: I will add it to the PR title though.

File: charts/rapid-response/values.yaml [hunk 3]
```diff
@@ -164,3 +164,19 @@ rapidResponse:
...
+
+rbac:
+  # true here enables creation of rbac resources
+  create: true
+
+serviceAccount:
+  # true here enables creation of service account
+  create: true
+  # Use this value as kspmCollectorServiceAccountName
+  name: "rapid-response"
```
Comment (dark-vex) [inline charts/rapid-response/values.yaml]: @yashgiri since these are test values, I think would be worth having them inside this file:
https://github.com/sysdiglabs/charts/blob/master/charts/rapid-response/ci/test-values.yaml.template

File: charts/rapid-response/README.md [hunk 4]
```diff
@@ -48,6 +48,9 @@ The following table lists the configurable parameters of the Sysdig Rapid Respon
 | `rapidResponse.sslVerifyCertificate`      | Set it to `false` for disabling the certificate verification                            | `true`                                                        |
 | `rapidResponse.tolerations`               | The tolerations for scheduling	                                                      | `node-role.kubernetes.io/master:NoSchedule`                   |
 |                                                                                                                                       `node-role.kubernetes.io/control-plane:NoSchedule`            |
+| `rbac.create`                             | If true, create & use RBAC resources                                                    | `true`   |
+| `serviceAccount.create`                   | Create serviceAccount                                                                   | `true`   |
+| `serviceAccount.name`                     | Use this value as serviceAccountName                                                    | ` `   |
```
Comment (dark-vex) [inline charts/rapid-response/README.md]: This one can be removed from the readme since we are not going to deploy any rbac/serviceaccount on customer clusters

**Review by dark-vex — CHANGES_REQUESTED** [review]
Added a few comments/changes

File: charts/rapid-response/values.yaml (hunk 3 above)
Comment (yashgiri) [reply charts/rapid-response/values.yaml]: oh yes ill remove this and put it at an appropriate place.

**Review by dark-vex — APPROVED** [review]
For rapid response chart it looks good to me now, thanks!

I will check if there is a better way to achieve the same goal without adding new keys under `tests` section of `values.yaml` but for now let's use this method :)
